core/templates/dev/head/components/forms/schema_editors/schema_based_list_editor_directive.html

```diff
@@ -25,7 +25,7 @@
 
   <!-- The margin-left is added here to left-align the button with the previous list items. -->
   <div style="height: 30px; margin-top: 4px;">
-    <button ng-show="!isOneLineInput || isAddItemButtonPresent"
+    <button ng-show="!isOneLineInput || isAddItemButtonPresent || (localValue.length && localValue[localValue.length - 1].length)"
```

Comment:
The 2-OR clauses are needed. The 2nd one adds the add item button when there are no input fields present. For Example: If there is only one input field and the input field is blurred then as empty input fields are removed by default then in that case we need an add item button which is also removed if we remove the second OR condition.